modernpython/langPack.py

```diff
@@ -87,9 +116,48 @@ def set_enum_classes(new_enum_classes):
 def set_float_classes(new_float_classes):
     return
 
+def has_unit_attribute(attributes):
+    for attr in attributes:
+        if attr["label"] == "unit":
+            return True
+    return False
 
 def run_template(version_path, class_details):
-    for template_info in template_files:
+    if (
+        # Primitives are never used in the in memory representation but only for
+        # the schema
+        (
+            class_details["class_name"]
+            in PRIMITIVE_CLASSES
+        )
+        # Datatypes based on primitives are never used in the in memory
+        # representation but only for the schema
+        or class_details["is_a_float"] == True
```

Comment:
@guillaume-alliander `<cims:stereotype>CIMDatatype</cims:stereotype>` are never used during the parsing of a cime file, this because they are just metadata.

For example, `ApparentPower` is used in datatype of the attribute `ACDCConverter.baseS`, which is just a float in the class model.

Options:
1. Given that are never used, we remove them as in this code.
2. We add to the Fields a metadata called `data_type` (in a similar fashion to `in_profile`) so that this metadata is preserved and can be used to extract units and scale factor.

Feelings?

Reply:
Maybe a tad late - I am of 2 minds. I do like the idea of preserving metadata for future use, because at the end of the day we are trying to be generic, but it might be an acute case of YAGNI. I think that if it low effort it's worth keeping it.

Reply:
generation of data types instances is done, I only need to attach them to the attributes, on tue i should be able to fix it.